src/shims/unix/solarish/foreign_items.rs

```diff
@@ -18,6 +19,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     ) -> InterpResult<'tcx, EmulateItemResult> {
         let this = self.eval_context_mut();
         match link_name.as_str() {
+            // Allocation
+            "memalign" => {
+                let [align, size] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let align = this.read_target_usize(align)?;
+                let size = this.read_target_usize(size)?;
+
+                // memalign requires the size to be greater than 0, the alignment to be a power of 2
+                // to be, at least, of word size in which EINVAL is emitted
+                // and a null pointer is returned.
+                // https://docs.oracle.com/cd/E88353_01/html/E37843/memalign-3c.html
+                if !align.is_power_of_two() || align < this.pointer_size().bytes() || size == 0 {
+                    this.set_last_error(this.eval_libc("EINVAL"))?;
```

Comment:
I don't think size 0 should lead to EINVAL.

Reply:
Ah, it is listed in the error conditions below: "or the size parameter is zero". But it's not listed above. Wow that's bad documentation.^^

Please point this out in the comments.